packages/ui/src/views/repo/pull-request/components/commit-selector/commit-selector.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, useMemo } from 'react'
+
+import { Button, DropdownMenu, DropdownMenuTrigger, Icon, Text } from '@/components'
+import { IBranchSelectorStore, TranslationStore, TypesCommit } from '@/views'
+
+import { CommitSelectorListItem } from '../../compare/components/types'
+import { CommitSelectorDropdown } from './commit-selector-dropdown'
+
+// import { BranchSelectorDropdown } from './branch-selector-dropdown'
+
+interface CommitSelectorProps {
+  useRepoBranchesStore: () => IBranchSelectorStore
+  useTranslationStore: () => TranslationStore
+  buttonSize?: 'default' | 'sm'
+  selectedCommit?: CommitSelectorListItem
+  onSelectCommit?: (commit: CommitSelectorListItem) => void
+  commitList?: TypesCommit[]
+}
+export const CommitSelector: FC<CommitSelectorProps> = ({
+  useRepoBranchesStore,
+  useTranslationStore,
+
+  buttonSize = 'default',
+  selectedCommit,
+  onSelectCommit,
+  commitList
+}) => {
+  const { repoId, spaceId } = useRepoBranchesStore()
```

Comment:
why are we using the branches store for getting repoid and spaceid? they should be coming from the url instead

Reply:
done